docs/src/components/templates.tsx

```diff
@@ -127,14 +117,18 @@ function useLastTruthyValue<T>(val: T): T {
 
 export default function Templates() {
   const router = useRouter();
-  const { template } = router.query;
+  const searchParams = useSearchParams();
+  const template = searchParams?.get('template');
+
   const templateMatch = templates.find(t => t.id === template);
 
   const lastTemplate = useLastTruthyValue(templateMatch);
 
   const onClose = () => {
-    const { template: _template, ...restQuery } = router.query;
-    router.push({ query: restQuery }, undefined, { scroll: false });
+    // Remove the template search param
+    const queryParams = new URLSearchParams(searchParams || '');
+    queryParams.delete('template');
+    router.push(`/?${queryParams.toString()}`);
```

Comment:
Should be able to remove the `.toString()` here since we're already stringifying with interpolation.

Reply:
Nice pickup